### PR TITLE
T dh 599b  Rest of standard object triggers

### PR DESCRIPTION
**Pull request by byarsawich**

### CODE REVIEW

File: src/classes/CreateTimeSheetEntries.cls [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+		List<TimeSheetEntry> tseList = new List<TimeSheetEntry>();
+
+		for(TimeSheet timeSheet: timeSheets.values()){
+			AssignedResource ar = (AssignedResource)resourceToAr.get(timeSheet.ServiceResourceId);
+			TimeSheetEntry tse = new TimeSheetEntry(
+					StartTime = Datetime.now(),
+					Status = 'New',
+					WorkOrderId = ar.ServiceAppointment.ParentRecordId,
+					Description = 'Auto Genereated Entry based on Service Appointment times.',
```
Comment (Fitchburgh) [inline src/classes/CreateTimeSheetEntries.cls]: Fix spelling

File: src/classes/ServiceAppointmentService.cls [hunk 2]
```diff
@@ -0,0 +1,350 @@
+/*
+ * Created by byarsawi on 8/31/2018.
+ */
+public class ServiceAppointmentService {
+
+	private static final String onHoldHex = '#18FFB4';
```
Comment (Fitchburgh) [inline src/classes/ServiceAppointmentService.cls]: change

File: src/classes/ServiceAppointmentService.cls [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+				sa.Asset__c = currWO.AssetId;
+				if (currWO.WorkType != null && currWO.StartDate != null) {
+					System.debug('Date Should Be Changed');
+					System.debug(currWO.StartDate);
+					sa.EarliestStartTime = currWO.StartDate;
+					//if there isn't an offest, default to 2 days.
+					if (currWO.Due_Date__c == null) {
+						Integer minuteOffset = currWO.WorkType.FSL__Due_Date_Offset__c != null
+											   ? Integer.valueOf(currWO.WorkType.FSL__Due_Date_Offset__c)
+											   : 2880;
```
Comment (Fitchburgh) [inline src/classes/ServiceAppointmentService.cls]: make constant

Comment (Fitchburgh) [reply src/classes/ServiceAppointmentService.cls]: magic number

File: src/classes/ServiceAppointmentService.cls [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+				WHERE Id IN :parentIds]);
+
+		Set<Id> workTypeIds = new Set<Id>();
+		workTypeIds.addAll((List<Id>) base_toc.ArrayUtils.getField(workOrders.values(), 'WorkTypeId'));
+		System.debug(workTypeIds);
+
+		Set<Id> assetIds = new Set<Id>();
+		assetIds.addAll((List<Id>) base_toc.ArrayUtils.getField(workOrders.values(), 'AssetId'));
+		Map<Id, Asset> assets = new Map<Id, Asset>(
+				[SELECT Id, UniqueObjectID__c
```
Comment (Fitchburgh) [inline src/classes/ServiceAppointmentService.cls]: should we rename this special field - Cara C.

File: src/classes/TimeSheetEntryService.cls [hunk 5]
```diff
@@ -3,23 +3,39 @@
...
-				FROM ProductConsumed
-				WHERE Truck_Driver__c IN :srIds AND WorkOrderId IN :woIDs];
-		//build a map of ServiceResourceId to Product Consumed for easier retrieval
-		Map<String, sObject> serviceToPC = base_toc.ArrayUtils.getMaptoField(allTruckPCs, 'Truck_Driver__c');
+		insert newPCs;
+	}
+
+	public static void updateTruckCost(TimeSheetEntry[] records, Map<Id,TimeSheetEntry> oldTSEMap) {
+		system.debug(records);
+		TSEDataWrapperClass tseWrapper = getIntiailData(records);
```
Comment (Fitchburgh) [inline src/classes/TimeSheetEntryService.cls]: fix name

File: src/classes/TimeSheetEntryService.cls [hunk 6]
```diff
@@ -35,21 +51,97 @@ public class TimeSheetEntryService {
...
+											  : currPC.QuantityConsumed + newQuant;
+					pcsToUpdate.put(currPC.Id, currPC);
+				}
 			}
 		}
 
 		update pcsToUpdate.values();
 	}
 
+	private static TSEDataWrapperClass getIntiailData(TimeSheetEntry[] records) {
```
Comment (Fitchburgh) [inline src/classes/TimeSheetEntryService.cls]: name

File: src/classes/UpdateAppointmentTimeSheets.cls [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+ */
+public class UpdateAppointmentTimeSheets {
+
+	public static void updateTimeSheets(Set<Id> saIds, Set<Id> woIDs){
+
+		Map<Id, AssignedResource> arMap = new Map<Id, AssignedResource>(
+				[SELECT Id, ServiceResourceId, ServiceAppointmentId, ServiceAppointment.ParentRecordId,
+				ServiceAppointment.Subject, ServiceResource.Name
+				FROM AssignedResource
+				WHERE ServiceAppointmentId IN :saIds AND ServiceResource.ResourceType != 'Crew']
```
Comment (Fitchburgh) [inline src/classes/UpdateAppointmentTimeSheets.cls]: does this work with 'crew' and not 'c'

File: src/classes/UpdateAppointmentTimeSheets.cls [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+				if (ts == null) {
+					//Not sure what to do here yet.
+					continue;
+				}
+				newTimeSheetEntries.add(new TimeSheetEntry(
+						StartTime = Datetime.now(),
+						EndTime = DateTime.now(),
+						Status = 'New',
+						WorkOrderId = ar.ServiceAppointment.ParentRecordId,
+						Description = 'Zero Time Entry created because user did not start work.',
```
Comment (Fitchburgh) [inline src/classes/UpdateAppointmentTimeSheets.cls]: change 'user' to 'resource'

File: src/triggers/WorkOrderTrigger.trigger [hunk 9]
```diff
@@ -1,5 +1,18 @@
-trigger WorkOrderTrigger on WorkOrder (before insert, before update, after insert, after update) {
+trigger WorkOrderTrigger on WorkOrder (before insert, before update, before delete, after insert, after update, after delete) {
```
Comment (Fitchburgh) [inline src/triggers/WorkOrderTrigger.trigger]: remove unnecessary trigger routes

File: src/classes/ServiceAppointmentService.cls (hunk 4 above)
Comment (byarsawich) [reply src/classes/ServiceAppointmentService.cls]: no

File: src/classes/CreateTimeSheetEntries.cls (hunk 1 above)
Comment (byarsawich) [reply src/classes/CreateTimeSheetEntries.cls]: fixed

File: src/classes/ServiceAppointmentService.cls (hunk 2 above)
Comment (byarsawich) [reply src/classes/ServiceAppointmentService.cls]: changed to #1E8449

File: src/classes/ServiceAppointmentService.cls (hunk 3 above)
Comment (byarsawich) [reply src/classes/ServiceAppointmentService.cls]: new shiny constant

File: src/classes/TimeSheetEntryService.cls (hunk 5 above)
Comment (byarsawich) [reply src/classes/TimeSheetEntryService.cls]: fixed name

File: src/classes/TimeSheetEntryService.cls (hunk 6 above)
Comment (byarsawich) [reply src/classes/TimeSheetEntryService.cls]: it's fixed

File: src/classes/UpdateAppointmentTimeSheets.cls (hunk 8 above)
Comment (byarsawich) [reply src/classes/UpdateAppointmentTimeSheets.cls]: changed copy

File: src/triggers/WorkOrderTrigger.trigger (hunk 9 above)
Comment (byarsawich) [reply src/triggers/WorkOrderTrigger.trigger]: removed unnecessary trigger contexts

File: src/classes/UpdateAppointmentTimeSheets.cls (hunk 7 above)
Comment (byarsawich) [reply src/classes/UpdateAppointmentTimeSheets.cls]: no it does not, changed to 'C'

File: src/classes/WorkOrders.cls [hunk 10]
```diff
@@ -16,14 +16,20 @@ global with sharing class WorkOrders extends Framework.Domain {
 		}
 	}
 
+	public override void applyDefaults() {
+		System.debug(LoggingLevel.INFO,'HELLO WORLD');
```
Comment (Fitchburgh) [inline src/classes/WorkOrders.cls]: prob should remove this - or not bc its a managed package, up to you.

File: src/classes/WorkOrders.cls [hunk 11]
```diff
@@ -16,14 +16,20 @@ global with sharing class WorkOrders extends Framework.Domain {
...
+	public override void applyDefaults() {
+		System.debug(LoggingLevel.INFO,'HELLO WORLD');
+	}
+
 	public override void beforeInsert() {
-		Framework.Log.push(Attachments.class.getName(),'beforeInsert');
+		System.debug(Framework.Log.CurrentLog());
+		Framework.Log.push(WorkOrders.class.getName(),'beforeInsert');
 		WorkOrderService.copyInfo((WorkOrder[]) records);
+		System.debug(LoggingLevel.INFO,'HELLO WORLD');
```
Comment (Fitchburgh) [inline src/classes/WorkOrders.cls]: another hello world debug

**Pull request merged**